src/Workspaces/Remote/ServiceHub/Host/ChecksumSynchronizer.cs

```diff
@@ -46,81 +44,59 @@ public async ValueTask SynchronizeSolutionAssetsAsync(Checksum solutionChecksum,
                     checksums.Remove(solutionChecksumObject.Checksum);
                     await _assetProvider.SynchronizeAssetsAsync(checksums, cancellationToken).ConfigureAwait(false);
                 }
+            }
 
-                // third and last get direct children for all projects and documents in the solution 
-                await SynchronizeProjectAssets_NoLockAsync(solutionChecksumObject.Projects, cancellationToken).ConfigureAwait(false);
+            // third and last get direct children for all projects and documents in the solution 
+            foreach (var project in solutionChecksumObject.Projects)
+            {
+                var projectStateChecksums = _assetProvider.GetRequiredAsset<ProjectStateChecksums>(project);
+                await SynchronizeProjectAssetsAsync(projectStateChecksums, cancellationToken).ConfigureAwait(false);
             }
         }
 
-        public async ValueTask SynchronizeProjectAssetsAsync(HashSet<Checksum> projectChecksums, CancellationToken cancellationToken)
+        public async ValueTask SynchronizeProjectAssetsAsync(ProjectStateChecksums projectChecksum, CancellationToken cancellationToken)
         {
             using (await s_gate.DisposableWaitAsync(cancellationToken).ConfigureAwait(false))
             {
-                await SynchronizeProjectAssets_NoLockAsync(projectChecksums, cancellationToken).ConfigureAwait(false);
+                await SynchronizeProjectAssets_NoLockAsync(projectChecksum, cancellationToken).ConfigureAwait(false);
             }
         }
 
-        private async ValueTask SynchronizeProjectAssets_NoLockAsync(IReadOnlyCollection<Checksum> projectChecksums, CancellationToken cancellationToken)
+        private async ValueTask SynchronizeProjectAssets_NoLockAsync(ProjectStateChecksums projectChecksum, CancellationToken cancellationToken)
```

Comment:
unlike before, this syncs just one project.  the caller of this is reponsible for handling multiple projects if that is relevant.